numpy/core/src/multiarray/mapping.c

```diff
@@ -1446,23 +1467,23 @@ array_subscript(PyArrayObject *self, PyObject *op)
     PyArrayObject *view = NULL;
     PyObject *result = NULL;
 
-    PyArrayMapIterObject * mit = NULL;
+    PyArrayMapIterObject *mit = NULL;
 
-    /* return fields if op is a string index */
+    /*
+     * If the provided arg is a string, it might be attempting to index
+     * a field in a structured array.
+     */
     if (PyDataType_HASFIELDS(PyArray_DESCR(self))) {
-        PyArrayObject *view;
-        int ret = _get_field_view(self, op, &view);
-        if (ret == 0){
-            if (view == NULL) {
-                return NULL;
-            }
-            return (PyObject*)view;
+        int status = _get_field_view(self, arg, &view);
+        if (status == 0) {
+            return (view == NULL) ? NULL : (PyObject *) view;
```

Comment:
```suggestion
            return (PyObject *) view;
```

Reply:
oh interesting, I can safely cast a null pointer?

Reply:
Yes; [even in C++](https://stackoverflow.com/a/10961864/102441), where the base class offset might not be zero, that's required to be legal.

Here `PyArrayObject` starts with the fields of `PyObject` anyway, so even that possibility isn't relevant.